src/Dapr.Client/InvokeClient.cs

```diff
@@ -0,0 +1,48 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// A client for interacting with the Dapr invoke endpoints.
+    /// </summary>
+    public abstract class InvokeClient
+    {
+        /// <summary>
+        /// Invokes a method using the Dapr invoke endpoints.
+        /// </summary>
+        /// <param name="serviceName">The name of the service to be called in the Dapr invoke request.</param>
+        /// <param name="methodName">THe name of the method to be called in the Dapr invoke request.</param>
+        /// <param name="data">The data to be sent within the body of the Dapr invoke request.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
+        /// <returns>A <see cref="Task" />.</returns>
+        public abstract Task InvokeMethodAsync(string serviceName, string methodName, string data, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Invokes a method using the Dapr invoke endpoints.
+        /// </summary>
+        /// <param name="serviceName">The name of the service to be called in the Dapr invoke request.</param>
+        /// <param name="methodName">THe name of the method to be called in the Dapr invoke request.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
+        /// <typeparam name="TValue">The data type.</typeparam>
+        /// <returns>A <see cref="Task" /> that will return a deserialized object of the reponse from the Invoke request. If the Dapr response content is null the return type will be null.</returns>
+        public abstract Task<TValue> InvokeMethodAsync<TValue>(string serviceName, string methodName, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Invokes a method using the Dapr invoke endpoints.
+        /// </summary>
+        /// <param name="serviceName">The name of the service to be called in the Dapr invoke request.</param>
+        /// <param name="methodName">THe name of the method to be called in the Dapr invoke request.</param>
+        /// <param name="data">The data to be sent within the body of the Dapr invoke request.</param>
```

Comment:
I am assuming the data to be the serialized json here.
I was thinking if we can make the apis look something like following, what do you think?
`
/// <summary>
        /// Invokes the specified method on target service, Request and Response data is serialized/deserialized using the default System.Text.Json Serialization.
        /// Request & Response can be serialized/deserialized by providing an optional delegate also if user wants custom serialization but not needed for this impl. can be added later if needed.
        /// </summary>
        public Task<TResponse> InvokeService<TRequest, TResponse>(string serviceId, string methodName, TRequest requestData, CancellationToken cancellationToken = default)
        {
            return default;
        }
`

Reply:
Yeah, I like that. passing in a strongly typed variable instead of just a random string that could really be anything feels a lot more robust.